### PR TITLE
feat: semantic search guardrails

### PR DESCRIPTION
**Pull request by michael-brennan2005**
Adds openai moderation api + discarding low score results. Also fixes GetClubs filter error.

### CODE REVIEW

File: backend/src/search/openai.go [hunk 1]
```diff
@@ -84,3 +84,65 @@ func (c *OpenAIClient) CreateEmbedding(items []Searchable) ([]Embedding, *errors
...
+type CreateModerationResponseBody struct {
+	Results []ModerationResult `json:"results"`
+}
+
+type ModerationResult struct {
+	Flagged bool `json:"flagged"`
+}
+
+func (c *OpenAIClient) CreateModeration(items []Searchable) ([]ModerationResult, *errors.Error) {
+	searchStrings := []string{}
```
Comment (garrettladley) [inline backend/src/search/openai.go:102]:
```suggestion
	searchStrings := make([]string, len(items))
```

File: backend/src/search/pinecone.go [hunk 2]
```diff
@@ -265,10 +276,12 @@ func (c *PineconeClient) Search(item Searchable, topK int) ([]string, *errors.Er
 		return []string{}, &errors.FailedToSearchToPinecone
 	}
 
-	resultsToReturn := make([]string, len(results.Matches))
+	resultsToReturn := []string{}
 
-	for i, match := range results.Matches {
-		resultsToReturn[i] = match.Id
+	for _, match := range results.Matches {
+		if match.Score > 0.7 {
```
Comment (garrettladley) [inline backend/src/search/pinecone.go:282]: should we make this configurable, perhaps in one of the ./config/*.yml files?

File: backend/src/transactions/club.go [hunk 3]
```diff
@@ -46,9 +44,9 @@ func GetClubs(db *gorm.DB, pinecone search.PineconeClientInterface, queryParams
 
 	if queryParams.Search != "" {
 		clubSearch := models.NewClubSearch(queryParams.Search)
-		resultIDs, err := pinecone.Search(clubSearch, 10)
+		resultIDs, err := pinecone.Search(clubSearch, 25)
```
Comment (garrettladley) [inline backend/src/transactions/club.go:47]: should we hard code topK or make it configurable?

File: backend/tests/search_test.go [hunk 4]
```diff
@@ -197,6 +197,26 @@ func TestPineconeSearchWorks(t *testing.T) {
 			},
 		})
 
+	gock.New("https://api.openai.com").
```
Comment (garrettladley) [inline backend/tests/search_test.go:200]: goated from you 🐐

**Review by garrettladley — COMMENTED** [review]
nits, so goated @michael-brennan2005 keep up the great work 💪

**Review by Blais3D — COMMENTED** [review]
LGTM

**Comment by garrettladley** [comment]
> LGTM

thx @Blais3D

**Pull request merged**